orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/StartAwsCodeBuildTask.java

```diff
@@ -51,4 +69,74 @@ public TaskResult execute(@Nonnull Stage stage) {
     context.put("buildInfo", execution);
     return TaskResult.builder(ExecutionStatus.SUCCEEDED).context(context).build();
   }
+
+  private void appendProjectName(Map<String, Object> requestInput, String projectName) {
+    requestInput.put(PROJECT_NAME, projectName);
+  }
+
+  private void appendSource(Map<String, Object> requestInput, Artifact artifact) {
+    switch (artifact.getType()) {
+      case "s3/object":
+        requestInput.put(SOURCE_TYPE, "S3");
+        String s3Location = artifact.getReference();
+        requestInput.put(
+            SOURCE_LOCATION, s3Location.startsWith("s3://") ? s3Location.substring(5) : s3Location);
+        break;
+      case "git/repo":
+        requestInput.put(SOURCE_LOCATION, artifact.getReference());
+        if (artifact.getReference().matches("^(http(s)?://)github.com/(.*)$")) {
+          requestInput.put(SOURCE_TYPE, "GITHUB");
+        } else if (artifact.getReference().matches("^(http(s)?://)(.*@)?bitbucket.org/(.*)$")) {
+          requestInput.put(SOURCE_TYPE, "BITBUCKET");
+        } else {
+          throw new IllegalStateException("Only GitHub and BitBucket repositories are supported");
+        }
```

Comment:
No support for GitHub Enterprise or CodeCommit?  Could the user specify what provider the artifact is for?

Reply:
Yes CodeCommit should be added here. That was a miss. For GHE, there's no proper regex match that we can do. Unless we categorize all non GitHub/BitBucket/CodeCommit repos to GHE, there's no easy way to enable this. 

Reference for git/repo type artifact https://www.spinnaker.io/reference/artifacts/types/git-repo/

Reply:
I think GHE will be important for the typical Spinnaker user.  What about a separate field for users to specify the "CodeBuild source type"?  It can be non-required so you can do nice inferring logic here, but allow customers to specify when it's not clear.

Reply:
That's a good idea. I will add a source type field to let users specify type explicitly.